pkg/util/http.go

```diff
@@ -32,6 +33,15 @@ func ParseTraceID(r *http.Request) ([]byte, error) {
 }
 
 func hexStringToTraceID(id string) ([]byte, error) {
+	// The encoding/hex package does not handle non-hex characters.
+	// Catch a few common characters early.
+	nonHexChars := []string{" ", "-", "/", "_", "\\", "(", ")", "[", "]"}
```

Comment:
This changes the error message, but it doesn't introduce any new information, so I'm not sure it will help the user resolve the issue better.  I.e. the space at the end may still be easily overlooked, without showing the position of the invalid char.  

I was on board with doing a simple trim for unambiguous cases (like a trailing space).  If we want to go the route of erroring still, then let's switch from enumerating invalid chars, to ensuring the input only has valid chars: 0-9, a-f, A-F, and print the position.

Reply:
Thanks for the review.  Good points about catching only valid characters.  I've updated the PR to reflect this desire and point at the string positions of any bad characters if they are found in the trace ID.